todo-app/src/App.js

```diff
@@ -1,13 +1,98 @@
 import React, { Component } from 'react';
+import Task from './Task' 
+import TaskForm from './TaskForm'
+import {loadTodoItems, saveTodoItems} from './storage'
+
+
+export default class App extends Component {
+    state = {
+    todos: loadTodoItems(),
+    editingTodoID: null
+  }
+
+handleEditTodo = id => {
+  this.setState({editingTodoID: id});
+};
+
+handleCancleEdit = ()=> {
+    this.setState({editingTodoID: null});
+}
+
+
+setTodos(todos) {
+    this.setState({todos})
+    saveTodoItems(todos)
+}
+
+handleSaveEdit = (id , description) => {
+    this.setTodos(this.state.todos.map(todoItem => {
+        if (todoItem.id === id) {
+            this.setState({editingTodoID: null});
+            return {...todoItem, description};
+        }else {
+            return todoItem;
+        }
+    }))
+};
+toggleStatus = (todoID) =>{
+    this.setTodos(this.state.todos.map(todo =>{
+                if (todo.id === todoID){
+                    todo.done = !todo.done
+                }
+                return todo
+                
+            })
+      )
+    }
+
+
+ newTaskWithText(text){
+     const ids = this.state.todos.map(todo => todo.id)
+     return {
+         id: ids.length === 0 ? 1 : Math.max(...ids) + 1,
+         description: text,
+         done: false
+     };
+ }
+ handleTaskAdd = newTodo => {
+     const todo = this.newTaskWithText(newTodo);
+     this.setTodos([todo, ...this.state.todos]);
+ };
+
+handleRemoveTodo = (todoItem) => {
+     const updateTodos = this.state.todos;
+    updateTodos.splice(updateTodos.indexOf(todoItem), 1)
+    this.setState({todos: updateTodos});
+    saveTodoItems(updateTodos);
+    
+}
 
-class App extends Component {
   render() {
+      const{todos} = this.state
     return (
       <div>
-        Hello World!
+        <h1>Todo App</h1>
+         <TaskForm 
+           ontodoAdd = {this.handleTaskAdd}
+          />
+		<ul className='tasks'>
+         {todos.map((todoItem, i) =>(
+        <Task 
+        key = {todoItem.id} 
```

Comment:
Just a head's up, since your todo IDs are simple numbers, you might want to make your keys here something like `task-${todoItem.id}`, etc. Keys need to be unique across an entire app (or at least the current view), so if you had other items with numerical IDs, you might see a clash.